cli/src/error.rs

```diff
@@ -18,7 +18,7 @@ pub enum CliError {
 /// Errors resulting from incomplete or invalid user input
 #[derive(Debug)]
 pub struct InvalidInputError {
-    details: String,
+    pub details: String,
```

Comment:
According to the `From` implementations I don't think you should have to expose this.

Reply:
It keeps saying it's private.  Is there something I'm doing wrong or missing here?  If not, I'll write a from_str() method.